src/Microsoft.DotNet.Interactive.Jupyter.Tests/MessagePlayback.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Subjects;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Interactive.Jupyter.Messaging;
+using Microsoft.DotNet.Interactive.Jupyter.Protocol;
+using Message = Microsoft.DotNet.Interactive.Jupyter.Messaging.Message;
+
+namespace Microsoft.DotNet.Interactive.Jupyter.Tests;
+
+internal class MessagePlayback : IMessageTracker
+{
+    private readonly Subject<Message> _sentMessages = new();
+    private readonly Subject<Message> _receivedMessages = new();
+    private readonly ConcurrentQueue<Message> _processRequests = new();
+    private readonly List<Message> _playbackMessages = new();
+    private readonly CancellationTokenSource _cts = new();
+    public IObservable<Message> Messages => _receivedMessages;
+
+    public MessagePlayback(IReadOnlyCollection<Message> messages)
+    {
+        _playbackMessages.AddRange(messages);
+
+        // FIX: (MessagePlayback) make sure this is awaited
```

Comment:
Is this Todo still relevant. Seems like it might be helpful to gracefully terminate the task captured here on Dispose.

Reply:
This one is a long-running task that's intended to do continuous work in the background. It could use some redesign but awaiting it is actually the wrong approach. I'll remove the comment.